src/components/TokenForm/TokenSelectModal.tsx

```diff
@@ -99,3 +69,39 @@ const TokenSelectModal: FC<TokenSelectModalProps> = ({
 }
 
 export default TokenSelectModal
+
+type TokenSelectOptionProps = {
+  tokenAddress: Address
+  underlyigAssets?: Address[]
+  onClick: MouseEventHandler<HTMLDivElement>
+  onKeyDown: KeyboardEventHandler<HTMLDivElement>
+}
+
+const TokenSelectOption = forwardRef<HTMLDivElement, TokenSelectOptionProps>(
+  ({ tokenAddress, underlyigAssets, onClick, onKeyDown }, ref) => {
+    return (
+      <RadioGroup.Option
+        as="div"
+        ref={ref}
+        className="grid grid-cols-[auto,1fr] grid-rows-[1fr,auto] items-center gap-x-2 rounded-lg p-2 ui-checked:bg-white/80 ui-checked:text-pink-900 ui-not-checked:bg-black ui-not-checked:text-white ui-disabled:cursor-not-allowed ui-disabled:opacity-50 ui-not-disabled:cursor-pointer md:gap-x-3 md:p-3"
+        value={tokenAddress}
+        onClick={onClick}
+        onKeyDown={onKeyDown}
+      >
+        <div className="row-span-2 row-start-1">
+          <AssetLogoWithUnderlyings
+            className="h-9 w-9 drop-shadow md:h-10 md:w-10"
+            tokenAddress={tokenAddress}
+            underlyingAssets={underlyigAssets}
+          />
+        </div>
+        <h2 className="col-start-2 row-start-1 font-medium">
+          <AssetSymbol address={tokenAddress} />
```

Comment:
Simplified this to just use the existing components for displaying asset information. The hook to retrieve multiple tokens at once is no longer required.